xdsl/dialects/experimental/aie.py

```diff
@@ -0,0 +1,1056 @@
+from __future__ import annotations
+
+from collections.abc import Iterable
+from typing import Annotated
+
+from xdsl.dialects import memref
+from xdsl.dialects.builtin import (
+    AnyAttr,
+    AnyIntegerAttr,
+    ArrayAttr,
+    Block,
+    FlatSymbolRefAttr,
+    Float32Type,
+    IndexType,
+    IntegerAttr,
+    IntegerType,
+    NoneAttr,
+    Region,
+    Signedness,
+    StringAttr,
+    SymbolRefAttr,
+    i32,
+    i64,
+)
+from xdsl.dialects.memref import _MemRefTypeElement
```

Comment:
here I would just copy/paste the definition, and also make it private to your file, clients shouldn't need to import TypeVars

Reply:
The alternative is to import `AttributeInvT`